pulsar-client-tools/src/main/java/org/apache/pulsar/shell/PulsarShell.java

```diff
@@ -567,6 +602,14 @@ private static boolean isQuitCommand(String line) {
         return line.equalsIgnoreCase("quit") || line.equalsIgnoreCase("exit");
     }
 
+    private static boolean isExitModeCommand(String line) {
+        return line.equalsIgnoreCase("exitmode");
+    }
+
+    private static boolean isChangeModeCommand(String line) {
+        return line.equalsIgnoreCase("admin") || line.equalsIgnoreCase("client") || line.equalsIgnoreCase("config");
```

Comment:
seems like duplication. I'd assume that the ShellMode enum would have a status lookup method instead.

Reply:
for example https://www.baeldung.com/java-enum-values contains some advice how this could be handled.